leap_year.rb

```diff
@@ -0,0 +1,18 @@
+def year
+    print "What is the starting year? "
+    start_year = gets.chomp.to_i
+    
+    print "What is the ending year? "
+    end_year = gets.chomp.to_i
+
+    for year in start_year..end_year
+        if year % 400 == 0
+            puts "The #{year} is leap year"
+        elsif year % 4 == 0 && year % 100 != 0
+            puts "The #{year} is leap year"
+        else
+            puts "The #{year} is NOT leap year"
+        end
+    end
+end
+year()
```

Comment:
It is working pretty well. It would be nice to have some sort of validation that ensures the user types only numbers, because, if we add any type of string like "añsdjfdjdfka", when converted into integer, it becomes 0. Then something like this:

What is the starting year? kadjflka
What is the ending year? 4

would generate

The 0 is leap year
The 1 is NOT leap year
The 2 is NOT leap year
The 3 is NOT leap year
The 4 is leap year 

(Personal opinion) That input should return a message like "kadjflka is not a number".